### ISSUE TITLE
Performance on ARM64 (Raspberry Pi4b)

### ISSUE THREAD

**Issue by xeonixo** [3rob3/ImmichFrame#72]
### Version

1.0.1.0

### Deployment Method

Desktop (ARM64, Raspberry Pi4b)

### Description

The transition between images is not very smooth. Unfortunately, the view is not really nice.

The application is actually not rocket science and should also be able to be displayed smoothly by a Pi.

Desktop Environment is Xfce, OS is DietPi (Debian)

Overclocking to maximum will not change anything and runs as absolut energy saving underclocking. 

### Reproduction

- Install DietPi on Raspberry Pi4b (2GB)
- Install / Configure ImmichFrame
- Run ImmichFrame and see laggy transitions

### Expectations

Smooth transitions, like when running on Windows/Mac

### Configuration

```xml
<Settings>
        <ImmichServerUrl>https://xxx.xx</ImmichServerUrl>
        <ApiKey>xxxxx</ApiKey>
        <Interval>20</Interval>
        <DownloadImages>false</DownloadImages>
        <ShowMemories>false</ShowMemories>
        <RenewImagesDuration>20</RenewImagesDuration>
        <ShowClock>true</ShowClock>
        <ClockFontSize>92</ClockFontSize>
        <ClockFormat>HH:mm</ClockFormat>
        <ShowPhotoDate>true</ShowPhotoDate>
        <PhotoDateFontSize>40</PhotoDateFontSize>
        <PhotoDateFormat>dd.MM.yyyy</PhotoDateFormat>
        <ShowImageDesc>true</ShowImageDesc>
        <ImageDescFontSize>40</ImageDescFontSize>
        <WeatherApiKey>xxxxx</WeatherApiKey>
        <WeatherFontSize>40</WeatherFontSize>
        <UnitSystem>metric</UnitSystem>
        <!--Your latitude/longitude coordinates (ex. 40.7128,74.0060)-->
        <WeatherLatLong>XX.XX,XX.XX</WeatherLatLong>
        <Language>de</Language>
</Settings>
```

### Pre-Submission Checklist

- [X] This is a bug report and not a feature request

- [X] I have provided all of the required information to reproduce the bug (config, logs, etc.)

- [X] I have checked for related issues and checked the documentation

**Comment by xeonixo** [comment]
Here is a video of a transition

[https://youtu.be/-znCaUHiowo?si=_mTiET938aixfHPF](https://youtu.be/-znCaUHiowo?si=_mTiET938aixfHPF)

**Comment by 3rob3** [comment]
Yeah, something is not right there (that's worse than on my Pi 2). I noticed you are missing this from your XML, can you add it and verify if you still see the issue. It should be using the default of 1.0 if missing but still worth a try. Also try raising it to 3.0 and see if it is even worse.
<TransitionDuration>1.0</TransitionDuration>

**Comment by 3rob3** [comment]
Nevermind, i just tried removing that from mine and no difference. My Pi 5 stays under 10% total CPU, and that is running Immich server, ImmichFrame, and 4 or 5 other containers/services. Do you have another SD card you could try Raspberry Pi OS on instead of Diet Pi?

**Comment by xeonixo** [comment]
ugh. so much work went to sizing the inner frame from my screen. will try, give me some time

**Comment by 3rob3** [comment]
Is it possible your Pi is thermal throttling?
All reports we have gotten are Pi 3's and up run smoothly.

**Comment by xeonixo** [comment]
Running on 50-55°C / 122-132°F, thermal limit is at 70°C / 158°C

I installed passive copper cooling blocks

**Comment by xeonixo** [comment]
okay, i put just the basics togehter. its currently running better then on dietpi. ill try to fix the overscan and display rotation, wich is currently ignored

**Comment by xeonixo** [comment]
I probably found the problem, I had configured a non vc4-kms-v3d custom setting with changed resolution to fit my screen within Passepartout 😅 well now runs smoother.

**Comment by xeonixo** [comment]
I know, doesn't really fit in my issue, but,

Could you perhaps somehow make a "visible area" possible in the config? I'm currently having the problem that I can't get the image to fit the smaller frame size by hook or by crook. The limitations of X11 and xrandr are still there. I get it pushed into the right corner, but not smaller overall. Because it's a full-screen application, I can't simply tinker with a window size.

I imagine an offset from left, right, top, bottom in pixels. 

Unfortunately, the Linux board resources are already exhausted

**Comment by 3rob3** [comment]
You try this?
https://raspberrypi.stackexchange.com/questions/142512/how-to-get-overscan-working-on-raspi-4

**Comment by xeonixo** [comment]
Doesn't work with vc4-kms-v3d with x11, terminal is fine but not desktop environment

**Comment by 3rob3** [comment]
Do you need full desktop? Could try Lite.

**Comment by 3rob3** [comment]
We could make a configurable "margin" fairly easily. I.e. Margin="10,10,10,10" (Left,Right,Top,Bottom).

**Comment by xeonixo** [comment]
That would be absolutely awesome. I'll guess, that others might need that feature too 😅

**Comment by xeonixo** [comment]
@3rob3 could you maybe specify your solution without desktop environment? That would be really useful for my usecase, cause the desktop is more an unwanted feature 😅

**Comment by 3rob3** [comment]
In Raspberry Pi Imager select your device, then scrool down to Raspberry Pi OS (other), and select Raspberry Pi OS Lite (64 bit).
![image](https://github.com/3rob3/ImmichFrame/assets/156599986/ab13ea50-4bd9-4555-befb-04f52ceb0e25)

**Comment by xeonixo** [comment]
Closed due to misconfiguration of RPi.
